src/Renci.SshNet/Session.cs

```diff
@@ -1335,6 +1333,11 @@ private Message ReceiveMessage(Socket socket)
 
             _inboundPacketSequence++;
 
+            if (_inboundPacketSequence == uint.MaxValue && _isInitialKex)
+            {
+                throw new SshConnectionException("Inbound packet sequence number is about to wrap during initial key exchange.", DisconnectReason.KeyExchangeFailed);
+            }
```

Comment:
This piece of code is mirrored from https://github.com/hierynomus/sshj/pull/917/comtmits/94fcc960e0fb198ddec0f7efc53f95ac627fe083 and https://github.com/openssh/openssh-portable/commit/1edb00c58f8a6875fad6a497aa2bacf37f9e6cd5.
However, I don't quite understand the motivation.
We've already made sure the kex init is the first packet.

Reply:
I have the same feeling that it tries to ensure the integrity of key exchange, thought I can't think of a hack model here.